nativelib/src/main/resources/addrinfo.c

```diff
@@ -0,0 +1,236 @@
+#include <sys/socket.h>
+#include <sys/types.h>
+#include <netdb.h>
+#include <string.h>
+#include <arpa/inet.h>
+#include <netinet/in.h>
+#include <netinet/ip_icmp.h>
+#include <stdio.h>
+#include <stdbool.h>
+#include <stdlib.h>
+
+char* scalanative_host_to_ip(char* host) {
```

Comment:
Could you re-write that in Scala? I think that you should probably already have most of the bindings you need.

We usually try to keep our C code as short and simple as possible (just forwarders, and converters for `struct`s).